### PR TITLE
Homework2: Maria Filiptsova

### PR DESCRIPTION
**Pull request by MariaFiliptsova**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/homework2/task2.py [hunk 1]
```diff
@@ -14,10 +14,26 @@ def longest_word(str_):
...
     # write your code here
-    return ''  # write return value here
+    import string
+    str_ = str_.split()
+    longest_word = None
+    longest_word_length = 0
+    punctuation_symbols = string.punctuation
+    for current_word in str_:
+        current_word_length = len(current_word)
+        if current_word[-1] in punctuation_symbols:
```
Comment (belovmd) [inline src/homework2/task2.py]: strip?

File: src/homework2/task3.py [hunk 2]
```diff
@@ -14,10 +14,18 @@ def sub_string(str_):
     """
 
     # write your code here
-    return ''  # write return value here
+    line_without_space = str_.replace(" ", "")
+    new_line = []
```
Comment (belovmd) [inline src/homework2/task3.py]: why is it called line if it is a list?

File: src/homework2/task6.py [hunk 3]
```diff
@@ -13,10 +13,14 @@ def palindrom(n):
     """
 
     # write your code here
-    return ''  # write return value here
+    n = str(n)
+    if n == n[::-1]:
```
Comment (belovmd) [inline src/homework2/task6.py]: strings are forbidden. only numbers operations

File: src/homework2/task7.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import os
```
Comment (belovmd) [inline src/homework2/task7.py:2]: Missed tasks descriptions - failed sanity checks. 
We have to put tasks descriptions to file

File: src/homework2/task7.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+n = int(input('Enter your number\n'))
+if n % 2 != 0:
+    print("Weird")
+elif 2 <= n <= 5:
+    print("Not Weird")
+elif 6 <= n <= 20:
+    print("Weird")
+elif n > 20:
+    print("Not Weird")
+
```
Comment (belovmd) [inline src/homework2/task7.py:20]: why is docstring separated from the previous task with 1 blank line and from it's task with 2?

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework2.md):
- 16
-  18

**Comment by MariaFiliptsova** [comment]
All changes made according to the comments.

**Comment by belovmd** [comment]
8

**Pull request closed without merging**